internal/etcd/etcd.go

```diff
@@ -2,35 +2,224 @@ package etcd
 
 import (
 	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
 
-	etcdclient "go.etcd.io/etcd/client"
-	"go.etcd.io/etcd/version"
+	"github.com/blang/semver"
```

Comment:
small nit, there's an existing dependency on `github.com/coreos/go-semver` which should be used instead of another semver lib.